src/uu/mktemp/src/mktemp.rs

```diff
@@ -141,24 +141,29 @@ struct Options {
 }
 
 impl Options {
-    fn from(matches: &ArgMatches) -> Self {
-        let tmpdir = matches
-            .get_one::<PathBuf>(OPT_TMPDIR)
-            .or_else(|| matches.get_one::<PathBuf>(OPT_P))
-            .cloned();
-        let (tmpdir, template) = match matches.get_one::<String>(ARG_TEMPLATE) {
+    fn new(
+        directory: bool,
+        dry_run: bool,
+        quiet: bool,
+        tmpdir: Option<PathBuf>,
+        suffix: Option<String>,
+        treat_as_template: bool,
+        template: Option<String>,
+        tmpdir_flag: bool,
+    ) -> Self {
+        let (tmpdir, template) = match template {
             // If no template argument is given, `--tmpdir` is implied.
             None => {
                 let tmpdir = Some(tmpdir.unwrap_or_else(env::temp_dir));
                 let template = DEFAULT_TEMPLATE;
                 (tmpdir, template.to_string())
             }
             Some(template) => {
-                let tmpdir = if env::var(TMPDIR_ENV_VAR).is_ok() && matches.get_flag(OPT_T) {
+                let tmpdir = if env::var(TMPDIR_ENV_VAR).is_ok() && treat_as_template {
```

Comment:
Nushell has their own env var handling, so I think that what we expose should not depend on env vars. Basically what I think should happen is that this method should not have been changed, but the `Options` struct should be exposed, so that they can construct it however they want. Just like how we did it in `cp`: https://github.com/uutils/coreutils/blob/615b562b64f2d392975a697fa36748f5c2759f08/src/uu/cp/src/cp.rs#L224

The `mktemp` function would then also take the `Options` struct instead of all the separate parameters.